### PR TITLE
Add HI-VAE temperature annealing and inference defaults

### PR DESCRIPTION
**Pull request by xuxu-wei**
## Summary
- add HI-VAE temperature hyperparameters and annealed Gumbel-Softmax sampling during training
- use the annealed minimum temperature for inference and persist the new hyperparameters when saving/loading
- add a regression test covering the non-annealed (tau=1) configuration

## Testing
- pytest -q

------
https://chatgpt.com/codex/tasks/task_e_68cc056a1d9c8320954a2165749b43f1

### CODE REVIEW

File: suave/model.py [hunk 1]
```diff
@@ -121,6 +137,24 @@ def __init__(
...
+            if tau_min <= 0.0:
+                raise ValueError("tau_min must be positive for HI-VAE mode")
+            if tau_min > tau_start:
+                raise ValueError("tau_min cannot exceed tau_start")
+            if tau_decay < 0.0:
+                raise ValueError("tau_decay must be non-negative")
+            self._tau_start = float(tau_start)
+            self._tau_min = float(tau_min)
+            self._tau_decay = float(tau_decay)
+            self._inference_tau = float(tau_min)
```
[the remark this answers is not in the document]
Comment (xuxu-wei) [reply suave/model.py:156]: @codex check if this problem has solved.
